### ISSUE TITLE
vmagent: High frequent scrape retry rate on TCP Reset

### ISSUE THREAD

**Issue by Vladikamira** [VictoriaMetrics/VictoriaMetrics#1289]
**Describe the bug**
We are running Victoria-Metrics on k8s cluster, and scrape all pods by annotations.
Some Pods has many containers and multiple Ports to scrape, but not all of them have `/metrics` endpoints. 
When vmagent srape such containers it receive TCP Reset as an response from container to scrape which is fine, but vmagent make a retry right after this TCP Reset, without waiting till next scrape configured as `scrape_interval`.
In our case, it leads to about roughly 500rps ( one RTT is about 1-2 ms) to each failed container. It affect vmagent CPU resources as well as Network itself.

**To Reproduce**
scrape enpoint which reject requests by TCP Reset, no HTTP protocol should be implemented

**Expected behavior**
either not to retry on TCP Reset and wait till next scrape configured by `scrape_interval` or have configuration option for amount of retries

**Screenshots**

vmagent CPU usage
<img width="376" alt="Screenshot 2021-05-12 at 11 20 01" src="https://user-images.githubusercontent.com/5423271/117997304-23d0cf80-b343-11eb-80f5-8e0749e9e59a.png">

Prometheus CPU usage
<img width="376" alt="Screenshot 2021-05-12 at 11 20 46" src="https://user-images.githubusercontent.com/5423271/117997325-27645680-b343-11eb-8bb8-64b485b91247.png">

vm_promscrape_scrape_retries_total metric
<img width="1660" alt="Screenshot 2021-05-12 at 11 18 08" src="https://user-images.githubusercontent.com/5423271/117997191-0d2a7880-b343-11eb-941a-eb361185674e.png">

**Version**
```
/ # /vmagent-prod --version
vmagent-20210501-071234-tags-v1.59.0-0-gfb097ff77
```

**Used command-line flags**
vmagent
```
    Args:
      -promscrape.config=/config/scrape.yml
      -remoteWrite.tmpDataPath=/tmpData
      -remoteWrite.url=http://victoria-metrics-cluster-vminsert.victoria-metrics.svc.cluster.local:8480/insert/0/prometheus/
      -envflag.enable=true
      -envflag.prefix=VM_
      -loggerFormat=json
      -promscrape.maxScrapeSize=128MB
```

**Additional context**
here are traffic dumps for the same Pod/container and how it scraped by Victoria-Metrics and Prometheus

# Victoria Metrics
```
- scrape 1
12:22:59.210849 IP 10.10.10.1.42886 > 10.30.176.75.9000: Flags [S], seq 1301422837, win 42600, options [mss 1420,sackOK,TS val 457440562 ecr 0,nop,wscale 7], length 0
12:22:59.211004 IP 10.30.176.75.9000 > 10.10.10.1.42886: Flags [S.], seq 2252221744, ack 1301422838, win 43648, options [mss 1420,sackOK,TS val 2242139933 ecr 457440562,nop,wscale 7], length 0
12:22:59.211010 IP 10.10.10.1.42886 > 10.30.176.75.9000: Flags [.], ack 1, win 333, options [nop,nop,TS val 457440562 ecr 2242139933], length 0
12:22:59.211069 IP 10.10.10.1.42886 > 10.30.176.75.9000: Flags [P.], seq 1:193, ack 1, win 333, options [nop,nop,TS val 457440562 ecr 2242139933], length 192
12:22:59.211201 IP 10.30.176.75.9000 > 10.10.10.1.42886: Flags [.], ack 193, win 340, options [nop,nop,TS val 2242139933 ecr 457440562], length 0
12:22:59.211231 IP 10.30.176.75.9000 > 10.10.10.1.42886: Flags [R.], seq 1, ack 193, win 340, options [nop,nop,TS val 2242139933 ecr 457440562], length 0
- scrape 2
12:22:59.211372 IP 10.10.10.1.42904 > 10.30.176.75.9000: Flags [S], seq 1940255956, win 42600, options [mss 1420,sackOK,TS val 457440562 ecr 0,nop,wscale 7], length 0
12:22:59.211541 IP 10.30.176.75.9000 > 10.10.10.1.42904: Flags [S.], seq 2111262315, ack 1940255957, win 43648, options [mss 1420,sackOK,TS val 2242139933 ecr 457440562,nop,wscale 7], length 0
12:22:59.211550 IP 10.10.10.1.42904 > 10.30.176.75.9000: Flags [.], ack 1, win 333, options [nop,nop,TS val 457440562 ecr 2242139933], length 0
12:22:59.211598 IP 10.10.10.1.42904 > 10.30.176.75.9000: Flags [P.], seq 1:193, ack 1, win 333, options [nop,nop,TS val 457440562 ecr 2242139933], length 192
12:22:59.211740 IP 10.30.176.75.9000 > 10.10.10.1.42904: Flags [.], ack 193, win 340, options [nop,nop,TS val 2242139933 ecr 457440562], length 0
12:22:59.211830 IP 10.30.176.75.9000 > 10.10.10.1.42904: Flags [R.], seq 1, ack 193, win 340, options [nop,nop,TS val 2242139934 ecr 457440562], length 0
- scrape 3
12:22:59.212086 IP 10.10.10.1.42926 > 10.30.176.75.9000: Flags [S], seq 2417611466, win 42600, options [mss 1420,sackOK,TS val 457440563 ecr 0,nop,wscale 7], length 0
12:22:59.212254 IP 10.30.176.75.9000 > 10.10.10.1.42926: Flags [S.], seq 2828133969, ack 2417611467, win 43648, options [mss 1420,sackOK,TS val 2242139934 ecr 457440563,nop,wscale 7], length 0
12:22:59.212271 IP 10.10.10.1.42926 > 10.30.176.75.9000: Flags [.], ack 1, win 333, options [nop,nop,TS val 457440563 ecr 2242139934], length 0
12:22:59.212353 IP 10.10.10.1.42926 > 10.30.176.75.9000: Flags [P.], seq 1:193, ack 1, win 333, options [nop,nop,TS val 457440563 ecr 2242139934], length 192
12:22:59.212484 IP 10.30.176.75.9000 > 10.10.10.1.42926: Flags [.], ack 193, win 340, options [nop,nop,TS val 2242139934 ecr 457440563], length 0
12:22:59.212510 IP 10.30.176.75.9000 > 10.10.10.1.42926: Flags [R.], seq 1, ack 193, win 340, options [nop,nop,TS val 2242139934 ecr 457440563], length 0
```

# Prometheus
```
- scarape 1
12:25:54.076765 IP 10.30.132.2.58246 > 10.30.176.75.9000: Flags [S], seq 1288642671, win 42600, options [mss 1420,sackOK,TS val 2483535144 ecr 0,nop,wscale 7], length 0
12:25:54.077007 IP 10.30.176.75.9000 > 10.30.132.2.58246: Flags [S.], seq 3186965993, ack 1288642672, win 43648, options [mss 1420,sackOK,TS val 1034432966 ecr 2483535144,nop,wscale 7], length 0
12:25:54.077040 IP 10.30.132.2.58246 > 10.30.176.75.9000: Flags [.], ack 1, win 333, options [nop,nop,TS val 2483535144 ecr 1034432966], length 0
12:25:54.077168 IP 10.30.132.2.58246 > 10.30.176.75.9000: Flags [P.], seq 1:246, ack 1, win 333, options [nop,nop,TS val 2483535144 ecr 1034432966], length 245
12:25:54.077332 IP 10.30.176.75.9000 > 10.30.132.2.58246: Flags [.], ack 246, win 340, options [nop,nop,TS val 1034432967 ecr 2483535144], length 0
12:25:54.077378 IP 10.30.176.75.9000 > 10.30.132.2.58246: Flags [R.], seq 1, ack 246, win 340, options [nop,nop,TS val 1034432967 ecr 2483535144], length 0
- scarape 2
12:26:04.076695 IP 10.30.132.2.58674 > 10.30.176.75.9000: Flags [S], seq 2984962207, win 42600, options [mss 1420,sackOK,TS val 2483545143 ecr 0,nop,wscale 7], length 0
12:26:04.077030 IP 10.30.176.75.9000 > 10.30.132.2.58674: Flags [S.], seq 863673813, ack 2984962208, win 43648, options [mss 1420,sackOK,TS val 1034442965 ecr 2483545143,nop,wscale 7], length 0
12:26:04.077055 IP 10.30.132.2.58674 > 10.30.176.75.9000: Flags [.], ack 1, win 333, options [nop,nop,TS val 2483545144 ecr 1034442965], length 0
12:26:04.077254 IP 10.30.132.2.58674 > 10.30.176.75.9000: Flags [P.], seq 1:246, ack 1, win 333, options [nop,nop,TS val 2483545144 ecr 1034442965], length 245
12:26:04.077390 IP 10.30.176.75.9000 > 10.30.132.2.58674: Flags [.], ack 246, win 340, options [nop,nop,TS val 1034442966 ecr 2483545144], length 0
12:26:04.077401 IP 10.30.176.75.9000 > 10.30.132.2.58674: Flags [R.], seq 1, ack 246, win 340, options [nop,nop,TS val 1034442966 ecr 2483545144], length 0
```

**Comment by valyala** [comment]
@Vladikamira , could you build `vmagent` from the commit e6fda03e8f3fcd3d85d6c8cf1ab734c409de74b2 according to [these instructions](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether it fixes the issue frequent retry rate of scrape targets that don't support HTTP?

**Comment by Vladikamira** [comment]
it works! Thanks @valyala 🙏 
<img width="1589" alt="Screenshot 2021-05-13 at 14 37 51" src="https://user-images.githubusercontent.com/5423271/118120590-6dc3bf00-b3f0-11eb-94f9-5ee8c09d5855.png">

**Comment by valyala** [comment]
Great! Then the bugfix will be included in the next release of `vmagent`.

**Comment by valyala** [comment]
The bugfix is included in `vmagent` starting from [v1.60.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
